### PR TITLE
made more c++ like

### PR DESCRIPTION
**Pull request by mhatina**
Changes:
 * one class per one .cpp/.hpp
 * not using `using namespace std;` in header files
 * attributes are at the bottom of the class and always private or protected (never public)

Suggestions:
 * Use camelCase for naming (classes are already named this way)
 * constructors should be marked as `= default` or erased, if they do nothing

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -4,9 +4,13 @@ CMAKE_MINIMUM_REQUIRED (VERSION 2.8)
 
 ADD_COMPILE_OPTIONS (-std=c++11 -Wall -Wextra -Wno-unused-parameter -fPIC)
 
+set(CMAKE_MACOSX_RPATH 0) # to disable MAC OS warnings
 
 PROJECT (libdnf)
 
+if (NOT PYTHON_DESIRED)
+    set (PYTHON_DESIRED "2")
```
Comment (dmach) [inline CMakeLists.txt]: Let's make Python 3 default

File: libdnf/CMakeLists.txt [hunk 2]
```diff
@@ -8,11 +8,15 @@ INCLUDE_DIRECTORIES ("${PROJECT_BINARY_DIR}")
...
         repo.cpp
+        repodict.cpp
 )
 
 SET (LIBDNF_HEADERS
+        modulepackage.hpp
         package.hpp
         repo.hpp
+        repodict.hpp
+        rpmpackage.hpp
```
Comment (dmach) [inline libdnf/CMakeLists.txt]: If we're going to split sources and headers this way, we need to move them into subdirs.
We should consider creating top-level header files for easier includes.

File: libdnf/package.hpp [hunk 3]
```diff
@@ -1,33 +1,16 @@
 #ifndef DNF_PACKAGE_HPP
 #define DNF_PACKAGE_HPP
 
-
 #include <string>
-using namespace std;
```
Comment (dmach) [inline libdnf/package.hpp]: We should really consider removing 'using namespace std' in header files. +1

File: libdnf/repo.cpp [hunk 4]
```diff
@@ -1,42 +1,26 @@
...
-#include <iostream>
 
 #include "repo.hpp"
 
 using namespace std;
 
 
 void Repo::set_repoid(string value) {
-    this->_repoid = value;
+    this->repoid = value;
```
Comment (dmach) [inline libdnf/repo.cpp]: I'd prefer underscore for private and protected members, especially when used without this-> (as jrohel suggested).

File: libdnf/repo.cpp [hunk 5]
```diff
@@ -1,42 +1,26 @@
...
 }
 
 
-const string& Repo::get_repoid() {
-    return this->_repoid;
+const string &Repo::get_repoid() const {
```
Comment (dmach) [inline libdnf/repo.cpp]: what the second const mean?

**Review by dmach — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (mhatina) [reply CMakeLists.txt]: Agree. It was copy-pasted from dnf.

File: libdnf/CMakeLists.txt (hunk 2 above)
Comment (mhatina) [reply libdnf/CMakeLists.txt]: Agree.

File: libdnf/repo.cpp (hunk 5 above)
Comment (mhatina) [reply libdnf/repo.cpp]: Second const marks function. It means it won't and it can't change the class.

File: libdnf/repo.cpp (hunk 4 above)
Comment (mhatina) [reply libdnf/repo.cpp]: Strongly disagree. Classes should be small enough so the attributes are unique and easily distinguishable. If the classes are so big that there is need of `_`, then those classes are violating Single responsibility principle. Also there are lots of IDEs that highlight the attributes.

Besides, you will quickly learn to ignore the `_` to see just a name of attribute and `_` will become just unseen clutter that we don't want.

Comment (mhatina) [reply libdnf/repo.cpp]: Really bad example I found googling:
```
class A {
    void foo(int age_) { // parameter
        int age = 18; // local scope
        if (age_ > age) {
            cout << legal << endl;
            _age = age_;
        }
    }
    int _age; // member
};
```

After few go-throughs, you will only see 5 `age`s, that looks the same, but are different. And if you comment, that you can see the `_` (of course you are focusing on seeing the `_`), try to read the code in few days without thinking what this is about and try to understand what the code does. I guarantee that you will not see the `_`.

**Comment by dmach** [comment]
It doesn't compile.
Can you also incorporate "lost changes because of rebase" into other commits?

**Pull request merged**